frontend/src/metabase/lib/entities.js

```diff
@@ -259,12 +259,14 @@ export function createEntity(def) {
         );
 
         if (notify) {
-          if (notify.undo) {
-            // pick only the attributes that were updated
-            const undoObject = _.pick(
-              originalObject,
-              ...Object.keys(updatedObject || {}),
-            );
+          // pick only the attributes that were updated
+          const undoObject = _.pick(
+            originalObject,
+            ...Object.keys(updatedObject || {}),
+          );
+          // https://github.com/metabase/metabase/pull/20874#pullrequestreview-902384264
+          const canUndo = !hasCircularReference(undoObject);
```

Comment:
@daltojohnso This is kind of a hack to not show an `undo` button when we detect circular reference.

https://www.loom.com/share/b60b9abe30ed490286636089ad77a110